### DISCUSSION TITLE
different alignment result between 10x-sc-xcr-vdj and 10x-vdj-bcr-full-length presets

### DISCUSSION THREAD

**Discussion by bigcat1001** [milaboratory/mixcr#1314] [Q&A] [answered]
I ran the same data with mixcr 4.1.2 and 4.4.2. However, I got huge different result.In 4.1.2,the alignment report is:
> Analysis date: Wed Aug 23 13:47:50 CST 2023
Input file(s): R1.clean.reads.fastq,R2_clean_add_cellbarcode_reads.fastq
Output file(s): ./sample_result.vdjca
Version: 4.1.2; built=Mon Nov 28 17:16:06 CST 2022; rev=ec9c6fd320; lib=repseqio.v2.0
Command line arguments: align --report ./sample_result.align.report.txt --json-report ./sample_result.align.report.json --threads 40 --preset 10x-vdj-bcr-full-length --species hsa --tag-pattern ^(CELL:N{12})(UMI:N{16})\^(R2:*) R1.clean.reads.fastq R2_clean_add_cellbarcode_reads.fastq ./sample_result.vdjca
Analysis time: 2.32m
Total sequencing reads: 10820540
Successfully aligned reads: 10478652 (96.84%)
Alignment failed, no hits (not TCR/IG?): 341888 (3.16%)
Overlapped: 0 (0%)
Overlapped and aligned: 0 (0%)
Alignment-aided overlaps: 0 (NaN%)
Overlapped and not aligned: 0 (0%)
No CDR3 parts alignments, percent of successfully aligned: 9935334 (94.82%)
Partial aligned reads, percent of successfully aligned: 371177 (3.54%)
TRA chains: 31 (0%)
TRA non-functional: 0 (0%)
TRB chains: 3 (0%)
TRB non-functional: 0 (0%)
TRG chains: 15 (0%)
TRG non-functional: 0 (0%)
IGH chains: 19 (0%)
IGH non-functional: 0 (0%)
TRAD chains: 3 (0%)
TRAD non-functional: 0 (0%)
IGK chains: 29 (0%)
IGK non-functional: 0 (0%)
IGL chains: 10478552 (100%)
IGL non-functional: 1755 (0.02%)
Realigned with forced non-floating bound: 0 (0%)
Realigned with forced non-floating right bound in left read: 0 (0%)
Realigned with forced non-floating left bound in right read: 0 (0%)
Tag parsing report:
  Execution time: 0ns
  Total reads: 10820540
  Matched reads: 10820540 (100%)
  Projection +R1 +R2: 10820540 (100%)
  For variant 0:
    For projection IntListKey(list=[1, 2]):
      CELL:Left position: 0
      UMI:Left position: 12
      CELL:Right position: 12
      UMI:Right position: 28
      R2:Left position: 0
      Variants: 0
      Cost: 0
      CELL length: 12
      UMI length: 16
      R2 length: 157
  ======================================

however,when I put the same data to 4.4.2,the report is :

> Analysis date: Wed Aug 23 15:40:40 CST 2023
Input file(s): R1.clean.reads.fastq,R2_clean_add_cellbarcode_reads.fastq
Output file(s): OUTPUT.vdjca
Version: 4.4.2; built=Sun Jul 30 16:17:59 CST 2023; rev=cd7ea52e83; lib=repseqio.v3.0.1
Command line arguments: align --report OUTPUT.align.report.txt --json-report OUTPUT.align.report.json --threads 40 --preset 10x-sc-xcr-vdj --species hsa --tag-pattern ^(CELL:N{12})(UMI:N{16})\^(R2:*) --reset-whitelist CELL R1.clean.reads.fastq R2_clean_add_cellbarcode_reads.fastq OUTPUT.vdjca
Analysis time: 2.07m
Total sequencing reads: 10820540
Successfully aligned reads: 60661 (0.56%)
Coverage (percent of successfully aligned):
  CDR3: 0%
  FR3_TO_FR4: 0%
  CDR2_TO_FR4: 0%
  FR2_TO_FR4: 0%
  CDR1_TO_FR4: 0%
  VDJRegion: 0%
Alignment failed: no hits (not TCR/IG?): 10534805 (97.36%)
Alignment failed: low total score: 225074 (2.08%)
Overlapped: 0 (0%)
Overlapped and aligned: 0 (0%)
Overlapped and not aligned: 0 (0%)
Alignment-aided overlaps, percent of overlapped and aligned: 0 (NaN%)
No CDR3 parts alignments, percent of successfully aligned: 60469 (99.68%)
Partial aligned reads, percent of successfully aligned: 192 (0.32%)
Realigned with forced non-floating bound: 0 (0%)
Realigned with forced non-floating right bound in left read: 0 (0%)
Realigned with forced non-floating left bound in right read: 0 (0%)
TRA chains: 146 (0.24%)
TRA non-functional: 0 (0%)
TRB chains: 13031 (21.48%)
TRB non-functional: 0 (0%)
TRD chains: 1 (0%)
TRD non-functional: 0 (0%)
TRG chains: 636 (1.05%)
TRG non-functional: 0 (0%)
IGH chains: 3262 (5.38%)
IGH non-functional: 0 (0%)
TRAD chains: 2853 (4.7%)
TRAD non-functional: 0 (0%)
IGK chains: 4043 (6.66%)
IGK non-functional: 0 (0%)
IGL chains: 36689 (60.48%)
IGL non-functional: 0 (0%)
Trimming report:
  R1 reads trimmed left: 4 (0%)
  R1 reads trimmed right: 17 (0%)
  Average R1 nucleotides trimmed left: 1.5341193692736222E-5
  Average R1 nucleotides trimmed right: 5.646668280880621E-5
Tag parsing report:
  Execution time: 0ns
  Total reads: 10820540
  Matched reads: 10820540 (100%)
  Projection +R1 +R2: 10820540 (100%)
  For variant 0:
    For projection +R1 +R2:
      CELL:Left position: 0
      UMI:Left position: 12
      CELL:Right position: 12
      UMI:Right position: 28
      R2:Left position: 0
      Variants: 0
      Cost: 0
      CELL length: 12
      UMI length: 16
      R2 length: 157
======================================

I don't know what happened.Can anyone help me?
Beside,the reason i switch 4.1.2 to 4.4.2 is I want to use --reset-whitelist CELL to skip cell whitelist step.However,when i used the 4.2.0,set `--reset-whitelist CELL`,it report “Unknown option: '--reset-whitelist'”. So how should i set --reset-whitelist ?

**Comment by mizraelson** [comment]
Hi, can you please share the exact commands that you used for both versions? Also, can you tell me more about the data you are using? The fastq files seem to have been modified, judging by the filenames.

Sincerely,
Mark

**Comment by bigcat1001** [comment]
4.1.2:
`mixcr analyze --verbose 10x-vdj-bcr-full-length --threads 40 \
--species hsa \
--tag-pattern "^(CELL:N{12})(UMI:N{16})\^(R2:*)" \
R1.clean.reads.fastq \
R2_clean_add_cellbarcode_reads.fastq \
sample_result`

4.4.2:
`mixcr analyze --verbose --preset 10x-sc-xcr-vdj --threads 40 \
--species hsa \
--reset-whitelist CELL \
--tag-pattern "^(CELL:N{12})(UMI:N{16})\^(R2:*)" \
R1.clean.reads.fastq \
R2_clean_add_cellbarcode_reads.fastq \
OUTPUT`
In 4.4.2,I also tried a command without --reset-whitelist CELL  and got the same result.



YES,I used modified fastq files, I indexed the reads names. The cell barcode is modified so the read length of R2 is 157bp,longer than 150.
The top 3 reads of R2 file is:
@0
CGGACTAGACCCTTCATCCACTCGATTTGGGGGGGTCCCTGATCGCTTCTCTGGCTCCAAGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAAGATGAGGCTGATTACTACTGCTCCTCGTTTGCAGGCAATAACCCCCTAGTC
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFF,FF:FFFFFFFFFFF:FFF:FFFFF,FFFF:FFFFFFFFFFFFFFFF,FFFFFFFF:FFFFFFFFFFFFFF::FF:,FFFFFFF:F:FFFF:FFFF
@1
TTCTTCGGACAATCATTCGTCTTATTTTGGGGGGGGGTCACAAGAGGCAGCGCTCTCGGGACGTCTCCACCATGGCCTGGGCTCTGCTGCTCCTCACCCTCCGCCCTCAGGTCACAGGGGCCTGGGCCCCCGCTGCCCGGACGCAGCCGGCCGCCGG
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFF:F,F,,FFF:F:,FFF,,FFFFFFF:,FFFF:FF::FF:FFFFFF,F,FFF:FFFFFF,FFF,FF,FFF,F:FF,F,F,FFF,:,FFFFF:F,F,F,FFFF,,,F,:,:FF::FF:FFFF
@2
TCACCATGGAAGGAGTTTTATTTAACCTGGGGGGGTTTCTAGTCGCTTCTCTGGCTCCAAGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTAATTATTACTGCAGCTCATATACAAGCAGCAACAGTTATGTC
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFF:FFFFFFFFF,FFF:FFFFFFFFFFFFFFFF:FF:,F:FFFF,F:FFFF,FFF:,F:F:FFFFFFFFFFFFF,FFFFFFFFF,FFFFFFFF,FF::FFFFFF:,:FF,:FF

**Comment by bigcat1001** [comment]
Anther question,what is the exact option of --reset-whitelist in 4.2.0?
I used `--reset-whitelist CELL` ,however,it always report “Unknown option: '--reset-whitelist'”.

**Comment by mizraelson** [comment, marked as answer]
Did you happen to reverse complement the R2 reads? By default, 10x R2 reads are reversed, and the preset is optimized for this orientation to speed up the analysis. This optimization was added with a recent update.
Try adding the following parameter:
`-Malign.parameters.readsLayout=Opposite`

MiXCR v4.1.2 didn't have the `--reset-whitelist` parameter.

**Reply by bigcat1001** [reply]
Hi,thanks for your help.`-Malign.parameters.readsLayout=Opposite` works.
By the way,I mean in 4.2.0,I used `--reset-whitelist CELL` ,but it reported “Unknown option: '--reset-whitelist'”.

**Reply by mizraelson** [reply]
Yes, although help message had this parameter it was only included later.
